index.d.ts

```diff
@@ -637,18 +637,14 @@ setTimeout(() => {
 }, 1000);
 ```
 */
-export function execa(
+export function execa<EncodingType extends EncodingOption = DefaultEncodingOption>(
 	file: string,
 	arguments?: readonly string[],
-	options?: Options
-): ExecaChildProcess;
-export function execa(
-	file: string,
-	arguments?: readonly string[],
-	options?: Options<BufferEncodingOption>
-): ExecaChildProcess<Buffer>;
-export function execa(file: string, options?: Options): ExecaChildProcess;
-export function execa(file: string, options?: Options<BufferEncodingOption>): ExecaChildProcess<Buffer>;
+	options?: Options<EncodingType>
+): ExecaChildProcess<EncodingType extends DefaultEncodingOption ? string : Buffer>;
```

Comment:
Should we create a type for `EncodingType extends DefaultEncodingOption ? string : Buffer` since it is used multiple times?

Reply:
I thought about that but decided to keep that PR simple. At the next step we already should create complicated type set to infer results. But if you want, I can add something like `GetStdoutStderrType<EncodingType extends EncodingOption>`

Reply:
> But if you want, I can add something like GetStdoutStderrType<EncodingType extends EncodingOption>

Yes, that sounds good. I think using such a type would actually simplify the PR, by making it shorter and with fewer `extends ? :` statements.

Reply:
Ok, it shouldn't take too much time. But plz be ready to re-run those unstable tests )